### PR TITLE
move GCERegionalPersistentDisk feature from cloud-provider into kube_features.go

### PR DESCRIPTION
**Pull request by logicalhan**
As of #77412, GCERegionalPersistentDisk is no longer used in cloud-provider. This PR removes the string declaration from cloud-provider and declares it directly in kube_features.go. 

This change prevents cloud-provider from bringing in apiserver and component-base (and csi-translation-lib from bringing those two things in transitively).

**What type of PR is this?**

/kind cleanup

```release-note
NONE
```
/sig cloud-provider

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -434,6 +433,12 @@ const (
 	//
 	// Enables the OpenStack Cinder in-tree driver to OpenStack Cinder CSI Driver migration feature.
 	CSIMigrationOpenStack utilfeature.Feature = "CSIMigrationOpenStack"
+
+	// owner: @verult
+	// GA: v1.13
+	//
+	// Enables the regional PD feature on GCE.
+	GCERegionalPersistentDisk utilfeature.Feature = "GCERegionalPersistentDisk"
```
Comment (andrewsykim) [inline pkg/features/kube_features.go]: Can we unexport this to prevent future usage or leave a comment?

Comment (logicalhan) [reply pkg/features/kube_features.go]: sure thing!

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -434,6 +433,12 @@ const (
 	//
 	// Enables the OpenStack Cinder in-tree driver to OpenStack Cinder CSI Driver migration feature.
 	CSIMigrationOpenStack utilfeature.Feature = "CSIMigrationOpenStack"
+
+	// owner: @verult
+	// GA: v1.13
+	//
+	// Enables the regional PD feature on GCE.
+	gceRegionalPersistentDisk utilfeature.Feature = "gceRegionalPersistentDisk"
```
Comment (andrewsykim) [inline pkg/features/kube_features.go]: The string value still has to be `GCERegionalPersistentDisk`

Comment (logicalhan) [reply pkg/features/kube_features.go]: fixed.

**Comment by lavalamp** [comment]
thanks for squashing!

/lgtm

**Comment by liggitt** [comment]
reported integration flake in https://github.com/kubernetes/kubernetes/issues/77573

/retest

**Pull request merged**